### DISCUSSION TITLE
Path data type

### DISCUSSION THREAD

**Discussion by tryoxiss** [nushell/nushell#12916] [Q&A] [answered]
I noticed that the [nushell data types](https://www.nushell.sh/book/types_of_data.html) contain lots of ones that are more domain specific like file sizes and durations, but there does not seem to be one for paths. This seems a bit odd, since paths are one of the main things you work with in a shell, much more then durations or file sizes, so its weird they get a data type and paths do not.

To be clear, its not really a problem as the path command works fine enough, I am more wondering why its done that way. It could just as easily be `"64mb" | size` just like how its `"./config" | path`.

**Comment by ghost** [comment, marked as answer]
path kind of exists: you can use `path` as type for a function and itll work as you expect.

for my personal opinion on why the current system is good:  
lets take a look at existing data types:
* date
  * storing date in a string would either have to enforce a standard format (+timezone, etc) and also make +1 day really hard
  * storing timestamps as a int would make it unreadable to humans
* size
  * you want maximum precision, therefore you would have to store it as bytes if it were a basic int, which is not human readable (you could also make everything gb, but then all my text files would be 0 size)
  * storing it as a string makes calculations impossible
* glob
  * would be possible to just store as string, but is nice this way
* boolean
  * usefull in a `if` and having js like auto casting for `true`, `yes`, etc would not be a great idea in my opinion
* path
  * having to cast it to use `path split` would be annoying
  * having a seperate version (or having to cast) for `str startswith` would be annoying
  * ...

**Reply by tryoxiss** [reply]
Ahhh ok, that makes sense. Still seems a tad weird, but I do get the rationale now. Thanks!
